### PR TITLE
Runner changes for doing step status async

### PR DESCRIPTION
**Pull request by raghavharness**
Runner changes to do execute step asynchronously

### CODE REVIEW

**Review by harshjain12 — APPROVED** [review]

File: command/harness/dlite/execute.go [hunk 1]
```diff
@@ -47,23 +55,63 @@ func (t *VMExecuteTask) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-	resp.DelegateMetaInfo.HostName = t.c.delegateInfo.Host
-	resp.DelegateMetaInfo.ID = t.c.delegateInfo.ID
-	httphelper.WriteJSON(w, resp, httpOK)
+	t.handleResponse(w, stepResp, distributed)
+}
+
+func (t *VMExecuteTask) handleResponse(w http.ResponseWriter, resp *api.PollStepResponse, distributed bool) {
+	if distributed {
+		var r VMTaskExecutionResponse
+		if resp.Error == "" {
```
Comment (vistaarjuneja) [inline command/harness/dlite/execute.go:102]: From the code, it seems that we are just returning an empty poll response from `HandleStep` in the case of async right? Is this block of code needed?

File: command/harness/dlite/execute.go [hunk 2]
```diff
@@ -47,23 +55,63 @@ func (t *VMExecuteTask) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		}
+	} else {
+		ctxState().Add(cancel, req.ExecuteVMRequest.StageRuntimeID, task.ID)
+	}
 
-	ctxState().DeleteTask(req.ExecuteVMRequest.StageRuntimeID, task.ID)
+	var stepResp *api.PollStepResponse
+	stepResp, err = harness.HandleStep(ctx, &req.ExecuteVMRequest, t.c.stageOwnerStore, &t.c.env, t.c.poolManager, t.c.metrics, distributed)
+	if !distributed {
+		ctxState().DeleteTask(req.ExecuteVMRequest.StageRuntimeID, task.ID)
```
Comment (vistaarjuneja) [inline command/harness/dlite/execute.go:87]: Nit: you can move this to L82 with a defer

File: command/harness/dlite/execute.go (hunk 1 above)
Comment (raghavharness) [reply command/harness/dlite/execute.go:102]: yes this is needed. Before poll we call the start step request as well. If some failure happens there then we need to handle that.

File: command/harness/dlite/execute.go (hunk 2 above)
Comment (raghavharness) [reply command/harness/dlite/execute.go]: Done

File: command/harness/step.go [hunk 3]
```diff
@@ -125,9 +134,15 @@ func HandleStep(ctx context.Context, r *ExecuteVMRequest, s store.StageOwnerStor
...
-		return nil, fmt.Errorf("failed to call LE.RetryPollStep: %w", err)
+	var pollResponse *api.PollStepResponse
+
+	if !async {
+		pollResponse, err = client.RetryPollStep(ctx, &api.PollStepRequest{ID: r.StartStepRequest.ID}, StepTimeout)
+		if err != nil {
+			return nil, fmt.Errorf("failed to call LE.RetryPollStep: %w", err)
+		}
+	} else {
+		pollResponse = &api.PollStepResponse{}
```
[the remark this answers is not in the document]
Comment (raghavharness) [reply command/harness/step.go]: Done

**Pull request merged**